learningequality/kolibri-design-system
# Storybook & some KButton stories

## Description

We've discussed in the past wanting to try out Storybook, particularly for KDS, and I had set it up with ease on a personal toy project a few months back so I have been optimistic about setting it up here and working with it.

- Depends on using the version of `kolibri-tools` in https://github.com/learningequality/kolibri/pull/9014 because Storybook requires Node >= 14 and `kolibri-tools` currently requires Node 10. 
- Initialized Storybook using `npx sb init` (very convenient, excellent install process)
- Configured the Storybook `preview.js` config to bootstrap `KThemePlugin` (may still need to do the same in the main.js config file to accommodate building Storybook - still more to learn here)
- Add dependencies for `sass` and `sass-loader` to devDependencies

## Steps to test

0) Be using Node 16
1) In local Kolibri, checkout this PR https://github.com/learningequality/kolibri/pull/9014
2) `cd` into `packages/kolibri-tools` and run `yarn link`
3) Go to your KDS directory, `yarn link kolibri-tools`
4) `yarn storybook` should spin up the server

## (optional) Implementation notes

### Does this introduce any tech-debt items?

*Story organization* is something I'm not sure about. Dumping them all in one folder may not be easy to track over a long period of time but that's a problem we can figure out when we get there I suppose.
